File: src/main/java/com/kingdom/model/PlayerTurn.java
package com.kingdom.model;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by IntelliJ IDEA.
 * Date: 5/3/11
 * Time: 8:16 PM
 */
public class PlayerTurn {
    private int userId;
    private String username = "";
    private List<String> history = new ArrayList<String>();

    public PlayerTurn(Player player) {
        this.userId = player.getUserId();
        this.username = player.getUsername();
        StringBuilder sb = new StringBuilder();
        sb.append("<span style='font-weight:bold'>");
        sb.append(username).append("'s Turn ").append(player.getTurns() + 1);
        history.add(sb.toString());
    }

    public void addHistory(String message) {
        history.add(message);
    }

    public List<String> getHistory() {
        return history;
    }

    public int getUserId() {
        return userId;
    }

    public String getUsername() {
        return username;
    }
}

File: src/main/kotlin/com/kingdom/model/PlayerTurn.kt
package com.kingdom.model

import com.kingdom.model.players.Player
import java.util.ArrayList

class PlayerTurn(player: Player) {
    val userId: String = player.userId
    var username = player.username
    val history = ArrayList<String>()
    val lastTurnSummary = player.lastTurnSummary

    @Suppress("unused")
    val reversedHistory
        get() = history.reversed()

    init {
        val sb = StringBuilder()
        sb.append("<span class='historyLabel'>")
        sb.append(username).append("'s Turn ").append(player.turns + 1)
        sb.append("</span>")
        history.add(sb.toString())
    }

    fun addHistory(message: String) {
        history.add(message)
    }
}
